src/update_agent/actor.rs

```diff
@@ -128,13 +128,9 @@ impl Handler<RefreshTick> for UpdateAgent {
                         .tick_end(&mut *agent_state_guard, update)
                         .await
                 }
-                UpdateAgentState::EndState => Ok(()),
+                UpdateAgentState::EndState => (),
```

Comment:
I think it could be possible to unify all the `.await` points above into a single one, perhaps after converting this one to an async closure.

Reply:
From https://users.rust-lang.org/t/expected-opaque-type-found-a-different-opaque-type-when-trying-futures-join-all/40596/4
Hmm so apparently "every `async` function generates a different, unique type, so we'll need to wrap the returned the future with `Pin<Box<dyn Future<Output = ()>>>`.
I think `.await` might still be the better option here.